src/core/providers/ogr/qgsogrprovider.cpp

```diff
@@ -7275,6 +7275,105 @@ QString QgsOgrProviderMetadata::getStyleById( const QString &uri, QString styleI
   return styleQML;
 }
 
+bool QgsOgrProviderMetadata::saveLayerMetadata( const QString &uri, const QgsLayerMetadata &metadata, QString &errorMessage )
+{
+  const QVariantMap parts = decodeUri( uri );
+  const QString path = parts.value( QStringLiteral( "path" ) ).toString();
+  if ( !path.isEmpty() )
+  {
+    QFileInfo fi( path );
+    if ( fi.suffix().compare( QLatin1String( "gpkg" ), Qt::CaseInsensitive ) == 0 )
+    {
+      const QString layerName = parts.value( QStringLiteral( "layerName" ) ).toString();
+      QgsOgrLayerUniquePtr userLayer;
+      userLayer = QgsOgrProviderUtils::getLayer( path, true, QStringList(), layerName, errorMessage, true );
+      if ( !userLayer )
+        return false;
+
+#if QT_VERSION < QT_VERSION_CHECK(5, 14, 0)
```

Comment:
Why is this check necessary? Is there any difference in QT versions regarding the use of a simple or recursive mutex?

Reply:
Since qt 5.14 the old flag approach to creating recursive mutexes was deprecated, in favour of the distinct QRecursiveMutex class. The older code still uses a recursive mutex, it just relies on code in the constructor to set the recursive flag.